### PR TITLE
fix(structured-list, pricing-table): correct coloration, spacing & alignment

### PR DESCRIPTION
**Pull request by andy-blum**
### Description

Corrects v1->v2 regressions in the structured-list and pricing-table components with cells containging icons.

Current:
Screenshot 2024-11-11 at 9 31 41 AM

Expected:
Screenshot 2024-11-11 at 9 31 12 AM

### Changelog

**Changed**

- Fixed spacing and alignment bugs
- Slight reworking of the `part` attributes

### CODE REVIEW

**Review by areagan1030 — APPROVED** [review]
Approve "cell with icon" display

**Comment by andy-blum** [comment]
Looks like the pricing-table is still wrong

https://deploy-preview-12101--ibm-dotcom-web-components.netlify.app/iframe.html?args=&knob-section%20heading=Optional%20section%20heading&knob-number%20of%20columns=3&knob-highlighted%20column=2&knob-highlighted%20label=Featured&knob-col-span-1=&knob-col-span-2=&knob-col-span-3=&knob-col-span-4=&knob-icon-text=lorem%20ipsum&id=experimental-pricing-table--default&viewMode=story

File: packages/styles/scss/components/structured-list/_structured-list.scss [hunk 1]
```diff
@@ -293,8 +293,16 @@
...
       color: $support-success;
     }
+
+    svg {
+      margin-inline-end: $spacing-03;
+    }
+  }
+
+  .#{$prefix}--structured-list-cell-icon-text-container {
+    display: flex;
```
Comment (m4olivei) [inline packages/styles/scss/components/structured-list/_structured-list.scss:306]: With this being `display: flex`, the `svg` flex item needs `flex-shrink: 0` to that it maintains its original size. As it is, if you have multiple lines of text for `icon-text`, the icon shrinks:

image

**Review by m4olivei — CHANGES_REQUESTED** [review]

File: packages/styles/scss/components/structured-list/_structured-list.scss (hunk 1 above)
Comment (andy-blum) [reply packages/styles/scss/components/structured-list/_structured-list.scss:307]: Should be fixed now:

Screenshot 2024-11-12 at 10 26 02 AM

**Review by m4olivei — APPROVED** [review]

**Comment by m4olivei** [comment]
@andy-blum There are some [annoying stylelint issues in the CI](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/actions/runs/11799023199/job/32866563616?pr=12101) 😐

**Pull request merged**